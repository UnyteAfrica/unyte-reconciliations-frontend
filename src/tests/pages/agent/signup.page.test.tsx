import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { AgentSignupPage } from "@/pages/agent/signup.page";

function setup(reactNode: React.ReactNode) {
  return {
    user: userEvent.setup(),
    ...render(reactNode),
  };
}

describe("Agent Signup Page", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should validate first name", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const nameInput = screen.getByLabelText("First Name");
    await user.click(nameInput);
    await user.keyboard("de");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "First name cannot be less than 3 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate middle name", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const nameInput = screen.getByLabelText("Middle Name");
    await user.click(nameInput);
    await user.keyboard("de");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Middle name cannot be less than 3 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate last name", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const nameInput = screen.getByLabelText("Last Name");
    await user.click(nameInput);
    await user.keyboard("de");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Last name cannot be less than 3 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate address", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const nameInput = screen.getByLabelText("Home Address");
    await user.click(nameInput);
    await user.keyboard("2 street");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Home Address cannot be less than 10 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate email", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const targetInput = screen.getByLabelText("Email");
    await user.click(targetInput);
    await user.keyboard("john@");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText("Invalid email entered");
    expect(errorMessage).toBeVisible();
  });

  it("should validate BVN", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const targetInput = screen.getByLabelText("BVN");
    await user.click(targetInput);
    await user.keyboard("12345678");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText("BVN must be 11 characters");
    expect(errorMessage).toBeVisible();
  });
  it("should validate bank name", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const targetInput = screen.getByLabelText("Bank Name");
    await user.click(targetInput);
    await user.keyboard("ba");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Bank name cannot be less than 3 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate account number", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const targetInput = screen.getByLabelText("Account No");
    await user.click(targetInput);
    await user.keyboard("12345");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Account number must be 10 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate affiliated company", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const targetInput = screen.getByLabelText("Affiliated Company");
    await user.click(targetInput);
    await user.keyboard("john");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Company name must be more than 5 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate GAMP ID", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const gampIdInput = screen.getByLabelText("GAMP ID (optional)");
    await user.click(gampIdInput);
    await user.keyboard("A099");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "GAMP ID cannot be less than 5 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate password", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const passwordInput = screen.getByLabelText("Password");
    await user.click(passwordInput);
    await user.keyboard("pass");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Password cannot be less than 6 characters"
    );
    expect(errorMessage).toBeVisible();
  });

  it("should validate confirm password", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentSignupPage />
      </BrowserRouter>
    );

    const passwordInput = screen.getByLabelText("Password");
    const confirmPasswordInput = screen.getByLabelText("Confirm Password");
    await user.click(passwordInput);
    await user.keyboard("password");
    await user.click(confirmPasswordInput);
    await user.keyboard("pass");
    const submitButton = screen.getByText("Sign Up");
    await user.click(submitButton);
    const errorMessage = screen.getByText("Passwords don't match");
    expect(errorMessage).toBeVisible();
  });
});
