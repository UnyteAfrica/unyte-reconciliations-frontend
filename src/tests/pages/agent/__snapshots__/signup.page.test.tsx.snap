// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Agent Signup Page > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="px-5 py-10 max-w-[600px] mx-auto"
        >
          <img
            alt=""
            class="mb-6 block"
            src="/src/assets/Icons/UnyteLogo.svg"
          />
          <form>
            <header>
              <p
                class="mb-2 font-semibold text-[20px]"
              >
                Agent Sign Up
              </p>
              <p
                class="mb-6 text-sm text-[#4F4F4F]"
              >
                Create your account by entering a username and password.
              </p>
            </header>
            <div
              class="space-y-6"
            >
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="firstName"
                >
                  First Name
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="firstName"
                  name="firstName"
                  placeholder="John"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="middleName"
                >
                  Middle Name
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="middleName"
                  name="middleName"
                  placeholder="Janet"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="lastName"
                >
                  Last Name
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="lastName"
                  name="lastName"
                  placeholder="Doe"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="homeAddress"
                >
                  Home Address
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="homeAddress"
                  name="homeAddress"
                  placeholder="2, Sizwe Street"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="email"
                >
                  Email
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="email"
                  name="email"
                  placeholder="johndoe@gmail.com"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="bankName"
                >
                  Bank Name
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="bankName"
                  name="bankName"
                  placeholder="Access Bank"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="accountNo"
                >
                  Account No
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="accountNo"
                  name="accountNo"
                  placeholder="1234567890"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="bvn"
                >
                  BVN
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="bvn"
                  name="bvn"
                  placeholder="12345678901"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="companyId"
                >
                  Company ID
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  disabled=""
                  id="companyId"
                  placeholder="A034529"
                  value=""
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div
                class=""
              >
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="gampId"
                >
                  GAMP ID
                   
                  <span
                    class="not-italic text-sm text-[#333]"
                  >
                    (optional)
                  </span>
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                  id="gampId"
                  name="gampId"
                  placeholder="A034529"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div>
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="flex relative items-center w-[100%] border rounded-lg transition duration-300 focus-within:border-black p-2 h-[58px] border-[#E0E0E0]"
                >
                  <div
                    class="w-full"
                  >
                    <input
                      class="block text-lg font-sora px-3 py-[16px] border-[#424242] rounded-lg outline-none mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0 h-[56px]"
                      id="password"
                      name="password"
                      placeholder="******"
                      type="password"
                      value=""
                    />
                    <p
                      class="text-sm text-[#333]"
                    />
                  </div>
                  <button
                    class="absolute top-[50%] -translate-y-[50%] right-3"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                      />
                    </svg>
                  </button>
                </div>
                <p
                  class="font-sora text-sm text-[#333]"
                />
              </div>
              <div>
                <label
                  class="mb-2 block text-sm text-[#333"
                  for="confirmPassword"
                >
                  Confirm Password
                </label>
                <div
                  class="flex relative items-center w-[100%] border rounded-lg transition duration-300 focus-within:border-black p-2 h-[58px] border-[#E0E0E0]"
                >
                  <div
                    class="w-full"
                  >
                    <input
                      class="block text-lg font-sora px-3 py-[16px] border-[#424242] rounded-lg outline-none mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0 h-[56px]"
                      id="confirmPassword"
                      name="confirmPassword"
                      placeholder="******"
                      type="password"
                      value=""
                    />
                    <p
                      class="text-sm text-[#333]"
                    />
                  </div>
                  <button
                    class="absolute top-[50%] -translate-y-[50%] right-3"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                      />
                    </svg>
                  </button>
                </div>
                <p
                  class="font-sora text-sm text-[#333]"
                />
              </div>
              <div>
                <button
                  class="w-full font-medium text-xl leading-[24px] bg-primary h-[72px] text-white rounded-2xl mb-4"
                >
                  Sign Up
                </button>
                <p
                  class="text-center"
                >
                  Already have an account?
                   
                  <a
                    class="text-primary"
                    href="/agent/login"
                  >
                    Sign In
                  </a>
                </p>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="px-5 py-10 max-w-[600px] mx-auto"
      >
        <img
          alt=""
          class="mb-6 block"
          src="/src/assets/Icons/UnyteLogo.svg"
        />
        <form>
          <header>
            <p
              class="mb-2 font-semibold text-[20px]"
            >
              Agent Sign Up
            </p>
            <p
              class="mb-6 text-sm text-[#4F4F4F]"
            >
              Create your account by entering a username and password.
            </p>
          </header>
          <div
            class="space-y-6"
          >
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="firstName"
              >
                First Name
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="firstName"
                name="firstName"
                placeholder="John"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="middleName"
              >
                Middle Name
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="middleName"
                name="middleName"
                placeholder="Janet"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="lastName"
              >
                Last Name
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="lastName"
                name="lastName"
                placeholder="Doe"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="homeAddress"
              >
                Home Address
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="homeAddress"
                name="homeAddress"
                placeholder="2, Sizwe Street"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="email"
              >
                Email
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="email"
                name="email"
                placeholder="johndoe@gmail.com"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="bankName"
              >
                Bank Name
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="bankName"
                name="bankName"
                placeholder="Access Bank"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="accountNo"
              >
                Account No
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="accountNo"
                name="accountNo"
                placeholder="1234567890"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="bvn"
              >
                BVN
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="bvn"
                name="bvn"
                placeholder="12345678901"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="companyId"
              >
                Company ID
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                disabled=""
                id="companyId"
                placeholder="A034529"
                value=""
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="gampId"
              >
                GAMP ID
                 
                <span
                  class="not-italic text-sm text-[#333]"
                >
                  (optional)
                </span>
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-[58px] border-[#E0E0E0]"
                id="gampId"
                name="gampId"
                placeholder="A034529"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div>
              <label
                class="mb-2 block text-sm text-[#333"
                for="password"
              >
                Password
              </label>
              <div
                class="flex relative items-center w-[100%] border rounded-lg transition duration-300 focus-within:border-black p-2 h-[58px] border-[#E0E0E0]"
              >
                <div
                  class="w-full"
                >
                  <input
                    class="block text-lg font-sora px-3 py-[16px] border-[#424242] rounded-lg outline-none mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0 h-[56px]"
                    id="password"
                    name="password"
                    placeholder="******"
                    type="password"
                    value=""
                  />
                  <p
                    class="text-sm text-[#333]"
                  />
                </div>
                <button
                  class="absolute top-[50%] -translate-y-[50%] right-3"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="font-sora text-sm text-[#333]"
              />
            </div>
            <div>
              <label
                class="mb-2 block text-sm text-[#333"
                for="confirmPassword"
              >
                Confirm Password
              </label>
              <div
                class="flex relative items-center w-[100%] border rounded-lg transition duration-300 focus-within:border-black p-2 h-[58px] border-[#E0E0E0]"
              >
                <div
                  class="w-full"
                >
                  <input
                    class="block text-lg font-sora px-3 py-[16px] border-[#424242] rounded-lg outline-none mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0 h-[56px]"
                    id="confirmPassword"
                    name="confirmPassword"
                    placeholder="******"
                    type="password"
                    value=""
                  />
                  <p
                    class="text-sm text-[#333]"
                  />
                </div>
                <button
                  class="absolute top-[50%] -translate-y-[50%] right-3"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="font-sora text-sm text-[#333]"
              />
            </div>
            <div>
              <button
                class="w-full font-medium text-xl leading-[24px] bg-primary h-[72px] text-white rounded-2xl mb-4"
              >
                Sign Up
              </button>
              <p
                class="text-center"
              >
                Already have an account?
                 
                <a
                  class="text-primary"
                  href="/agent/login"
                >
                  Sign In
                </a>
              </p>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;
